.travis.yml

```diff
@@ -33,9 +34,133 @@ matrix:
 before_script:
   - export PATH="$HOME/.composer/vendor/bin:$PATH"
   - |
+    ORIG_DIR=`pwd`
     if [[ ! -z "$WP_VERSION" ]] ; then
       bash bin/install-wp-tests.sh wordpress_test root '' localhost $WP_VERSION
-      if [[ ${TRAVIS_PHP_VERSION:0:2} == "5." ]]; then
+      if [[ ${SWITCH_TO_PHP:0:3} == "5.2" ]] || [[ ${SWITCH_TO_PHP:0:3} == "5.3" ]]; then
+        # install php runtime dependencies
+        sudo apt-get install -y curl libxml2 libfreetype6 libpng12-0 libjpeg8 libgd3 libxpm4 \
+        libltdl7 openssl gettext libgettextpo0 libmhash2 libmcrypt4 mysql-client
+
+        if [[ ${SWITCH_TO_PHP:0:3} == "5.2" ]]; then
+          PHPBREW_BUILT_CHECK=$HOME/.phpbrew/php/php-5.2.17/bin/php
+        else
+          PHPBREW_BUILT_CHECK=$HOME/.phpbrew/php/php-5.3.29/bin/php
+        fi
+
+        # php and phpunit installs should be cached, only build if they're not there.
+        if [ ! -f $PHPBREW_BUILT_CHECK ]; then
+          
+          # install build dependencies for building php (yes, for phpbrew, php is a dependency)
+          sudo apt-get install -y php5-dev autoconf automake curl libcurl3-openssl-dev build-essential \
+          libxslt1-dev re2c libxml2-dev php5-cli bison libbz2-dev libreadline-dev libfreetype6-dev \
+          libpng12-dev libjpeg-dev libjpeg8-dev libgd-dev libltdl-dev libssl-dev libgettextpo-dev \
+          libicu-dev libmhash-dev libmcrypt-dev libmysqlclient-dev libmysqld-dev git
+
+          # install phpbrew for this user
+          mkdir $HOME/php-utils-bin
+          cd $HOME/php-utils-bin
+          curl -L -O https://github.com/phpbrew/phpbrew/raw/master/phpbrew
+          chmod +x phpbrew
+
+          # init with known --old to get 5.2 and 5.3
+          $HOME/php-utils-bin/phpbrew init
+          $HOME/php-utils-bin/phpbrew known --old
+
+          # PHP5.2/3 require patches to compile, so download them
+          cd $HOME
+          mkdir patches
+          curl -L -o patches/node.patch https://mail.gnome.org/archives/xml/2012-August/txtbgxGXAvz4N.txt
+          curl -L -o patches/openssl.patch "https://bugs.php.net/patch-display.php?bug_id=54736&patch=debian_patches_disable_SSLv2_for_openssl_1_0_0.patch&revision=1305414559&download=1"
```

Comment:
Should we host them somewhere ourselves? Otherwise we'd have to duplicate them in core too. But that might not be a concern, I doubt we'll be updating them too often...